src/reducers/comments.tsx

```diff
@@ -203,18 +222,36 @@ const getKeyAndInfo = (state: CommentsState, keyParams: CommentKeyParams) => {
   return { key, info };
 };
 
+export const stateForVersion = ({
+  state,
+  versionId,
+}: {
+  state: CommentsState;
+  versionId: number;
+}): CommentsState => {
+  const reset = state.forVersionId !== versionId;
+  return {
+    ...state,
+    byId: reset ? {} : state.byId,
+    byKey: reset ? {} : state.byKey,
+    forVersionId: versionId,
+  };
+};
+
 const reducer: Reducer<CommentsState, ActionType<typeof actions>> = (
   state = initialState,
   action,
 ): CommentsState => {
   switch (action.type) {
     case getType(actions.beginComment): {
-      const { key, info } = getKeyAndInfo(state, action.payload);
+      const { versionId, ...keyParams } = action.payload;
+      const newState = stateForVersion({ state, versionId });
```

Comment:
Every single case contains these two statements:

```
const newState = stateForVersion({ state, versionId });
const { key, info } = getKeyAndInfo(newState, keyParams);
```

So I wonder if it would be worth having a helper that wraps both of these and returns `{ info, key, newState }`?

Reply:
Heh, yeah, I was thinking about that later on after requesting a re-review. I'll try and merge them.